### ISSUE TITLE
Some common buttons are represented by images and are not labeled for accessibility

### ISSUE THREAD

**Issue by nebeker** [ccrama/Slide#2983]
Slide version: 6.0.1-3
Android version: 9

Buttons in submissions are represented by images and are not labeled for accessibility. These include:

- Edit
- Hide
- Save
- Mod
- Upvote
- Downvote
- Menu

When using TalkBack, these buttons are all read aloud as "unlabeled button." This makes everyday use of the app by visual impaired users or others who wish, for whatever reason to use TalkBack, at best, depend on prior knowledge of the app's layout or, at worst, impossible.

These buttons can easily be labeled with content labels as per https://support.google.com/accessibility/android/answer/7158690?hl=en

**Comment by nebeker** [comment]
Hi @Tensounder54 I'm currently working on this and should only take a while to make a poll request. Is that OK or should I let somebody more experienced than myself handle it?

**Comment by Tensounder54** [comment]
Nope that's OK. Hope its easy to fix! Help is always appreciated. Just make sure you read the `Code` section of the README.MD file.

**Comment by nebeker** [comment]
As of now, the app is a lot more usable with TalkBack. While working on this I kept finding lots more unlabeled buttons - some are more important than others, but it's universally jarring to find one, because you'll never know whether or not it matters to you, because you don't know what it does.

I haven't managed to get buttons to animate on submissons and would appreciate some help with that. @Tensounder54 Is there any way to mark this issue as being worked on, but get some help on this?

Besides that point (and any other dynamic buttons) it's just a matter of time until I can add everything.

Moving forward, insuring accessibility for new features should be easy.

**Comment by Tensounder54** [comment]
It's marked as `Will Close On Merge` and `Help Needed`. These two will let people know that you need help and that you are working on this issue.

**Comment by nebeker** [comment]
Sounds good. Thanks.

**Comment by Tensounder54** [comment]
@nebeker Since you said you need help, what's it with? How can I help you?

**Comment by nebeker** [comment]
I'm still trying to get to grips with the structure of the app, so some things aren't working.

If you look at `app/src/main/java/me/ccrama/redditslide/Adapters/CommentAdapter.java`, this line works:

`upvote.setContentDescription(mContext.getResources().getString(R.string.btn_upvoted));`

However, in `app/src/main/java/me/ccrama/redditslide/SubmissionViews/PopulateSubmissionViewHolder.java`, a similar line doesn't seem to have any effect:

`upvotebutton.setContentDescription(mContext.getString(R.string.btn_upvoted));`

The upvote button chamges to "upvoted" in the comment menu, but not in submissions.

I'm wondering if maybe I'm changing the wrong file.

To test this you can go into your device's settings ->Accessibility and enable TalkBack. You can check the tutorial, but the basics are tap to select, double tap to activate and double tap and hold to hold.
